tools/packaging/static-build/cache_components.sh

```diff
@@ -74,10 +74,43 @@ cache_virtiofsd_artifacts() {
 	create_cache_asset "${virtiofsd_tarball_name}" "${current_virtiofsd_version}" "${current_virtiofsd_image}"
 }
 
+cache_rootfs_artifacts() {
+	# We need to remove `-dirty` from teh osbuilder_last_commit as the rootfs artefacts are generated on that folder
+	local osbuilder_last_commit="$(echo $(get_last_modification "${repo_root_dir}/tools/osbuilder") | sed s/-dirty//)"
+	local guest_image_last_commit="$(get_last_modification "${repo_root_dir}/tools/packaging/guest-image")"
+	local agent_last_commit="$(get_last_modification "${repo_root_dir}/src/agent")"
+	local libs_last_commit="$(get_last_modification "${repo_root_dir}/src/libs")"
+	local attestation_agent_version="$(get_from_kata_deps "externals.attestation-agent.version")"
+	local gperf_version="$(get_from_kata_deps "externals.gperf.version")"
+	local libseccomp_version="$(get_from_kata_deps "externals.libseccomp.version")"
+	local pause_version="$(get_from_kata_deps "externals.pause.version")"
+	local skopeo_version="$(get_from_kata_deps "externals.skopeo.branch")"
+	local umoci_version="$(get_from_kata_deps "externals.umoci.tag")"
+	local rust_version="$(get_from_kata_deps "languages.rust.meta.newest-version")"
+	local rootfs_tarball_name="kata-static-cc-rootfs-image.tar.xz"
+	local aa_kbc="offline_fs_kbc"
+	local image_type="image"
+	local root_hash_vanilla="${repo_root_dir}/tools/osbuilder/root_hash_vanilla.txt"
+	local root_hash_tdx=""
+	if [ -n "${TEE}" ]; then
+		if [ "${TEE}" == "tdx" ]; then
```

Comment:
What is the reason for these conditionals to be

Reply:
The only reason I'm using explicit conditionals as I consider this easier to read and easier to understand what are the parts that may need to be expanded in case a new TEE cache ends up being added (and I hope that will happen).

Reply:
Cool that makes sense and helps explain why `image_type="image"` is in the TDX block as it is default, but others (e.g. SEV) will need to update that to initrd. Thanks!